### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by nsmithuk** [chanzuckerberg/aurora-data-api#21]
I've started getting a strange error back from `aurora-data-api`. I'd be grateful to know if anyone else is seeing it. 

It seems that when a table not found exception is thrown, `aurora-data-api` is no longer catching it as expected.

So it is the case that Table 'xxx.alembic_version' does not exist. What would typically then happen in this situation is that `aurora-data-api` catches the error and it raises back up to `sqlalchemy`. However that no longer seems to be the case.

```
DEBUG [aurora_data_api] execute 'DESCRIBE `alembic_version`'
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/aurora_data_api/__init__.py", line 222, in execute
    res = self._client.execute_statement(**execute_statement_args)
  File "/usr/local/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.8/site-packages/botocore/client.py", line 676, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.BadRequestException: An error occurred (BadRequestException) when calling the ExecuteStatement operation: Database error code: 1146. Message: Table 'xxx.alembic_version' doesn't exist

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/alembic", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/alembic/config.py", line 581, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/usr/local/lib/python3.8/site-packages/alembic/config.py", line 575, in main
    self.run_cmd(cfg, options)
  File "/usr/local/lib/python3.8/site-packages/alembic/config.py", line 552, in run_cmd
    fn(
  File "/usr/local/lib/python3.8/site-packages/alembic/command.py", line 298, in upgrade
    script.run_env()
  File "/usr/local/lib/python3.8/site-packages/alembic/script/base.py", line 489, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/usr/local/lib/python3.8/site-packages/alembic/util/pyfiles.py", line 98, in load_python_file
    module = load_module_py(module_id, path)
  File "/usr/local/lib/python3.8/site-packages/alembic/util/compat.py", line 184, in load_module_py
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "alembic/env.py", line 87, in <module>
    run_migrations_online()
  File "alembic/env.py", line 81, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/usr/local/lib/python3.8/site-packages/alembic/runtime/environment.py", line 846, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/usr/local/lib/python3.8/site-packages/alembic/runtime/migration.py", line 498, in run_migrations
    heads = self.get_current_heads()
  File "/usr/local/lib/python3.8/site-packages/alembic/runtime/migration.py", line 435, in get_current_heads
    if not self._has_version_table():
  File "/usr/local/lib/python3.8/site-packages/alembic/runtime/migration.py", line 447, in _has_version_table
    return sqla_compat._connectable_has_table(
  File "/usr/local/lib/python3.8/site-packages/alembic/util/sqla_compat.py", line 61, in _connectable_has_table
    return connectable.dialect.has_table(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/dialects/mysql/base.py", line 2603, in has_table
    rs = connection.execution_options(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1003, in execute
    return self._execute_text(object_, multiparams, params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1172, in _execute_text
    ret = self._execute_context(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1514, in _handle_dbapi_exception
    util.raise_(exc_info[1], with_traceback=exc_info[2])
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/local/lib/python3.8/site-packages/aurora_data_api/__init__.py", line 232, in execute
    raise self._get_database_error(e) from e
  File "/usr/local/lib/python3.8/site-packages/aurora_data_api/__init__.py", line 209, in _get_database_error
    err_info = self._render_response(err_res)["records"][-1]
IndexError: list index out of range
```

It looks like to determine what the error was, it runs:
```
err_res = self._client.execute_statement(**self._prepare_execute_args("SHOW ERRORS"))
```

However the `records` array returned from SHOW ERRORS doesn't contain any errors.
```
{'ResponseMetadata': {'HTTPHeaders': {'content-length': '41',
                                      'content-type': 'application/json',
                                      'date': 'Thu, 29 Oct 2020 18:55:20 GMT',
                                      'x-amzn-requestid': 'e2178ed8-43c4-48f7-b401-46f2cfc57615'},
                      'HTTPStatusCode': 200,
                      'RequestId': 'e2178ed8-43c4-48f7-b401-46f2cfc57615',
                      'RetryAttempts': 0},
 'numberOfRecordsUpdated': 0,
 'records': []}
```

Then thus we get an IndexError.

**Comment by jurasofish** [comment]
Just started getting this too.

For me it's happening when trying to create tables using SQLAlchemy, fortunately I can work around it by setting checkfirst to False:
https://docs.sqlalchemy.org/en/13/core/metadata.html#sqlalchemy.schema.MetaData.create_all

**Comment by nsmithuk** [comment]
As a bit of an experiment I added a hack to skip over "SHOW ERRORS" for a Table Not Found, pulling the code out of the exception message. Everything carries on fine when I add this, leading me to think something has definitely changed around how those errors are returned.

https://github.com/nsmithuk/aurora-data-api/commit/d630c07dc78c3b821dec870488bec364e3cfd987

**Comment by nsmithuk** [comment]
This only seems to happen if you are running `Aurora (MySQL 5.7) 2.07.1`.
The problem doesn't occur with `Aurora (MySQL)-5.6.10a`.

**Comment by kislyuk** [comment]
Thanks for reporting, I have released a fix in v0.2.6. Please test and let me know if you continue to see any issues.
